packages/auth/amplify_auth_cognito/lib/src/model/signin/cognito_confirm_sign_in_options.dart

```diff
@@ -15,26 +15,26 @@
 
 import 'package:amplify_core/amplify_core.dart';
 
+/// {@template amplify_auth_cognito.model.cognito_confirm_sign_in_options}
+/// Cognito options for `Amplify.Auth.confirmSignIn`.
+/// {@endtemplate}
 class CognitoConfirmSignInOptions extends ConfirmSignInOptions {
-  final Map<String, String>? clientMetadata;
-  final Map<CognitoUserAttributeKey, String>? userAttributes;
-
+  /// {@macro amplify_auth_cognito.model.cognito_confirm_sign_in_options}
   const CognitoConfirmSignInOptions({
     this.clientMetadata,
     this.userAttributes,
   });
 
-  Map<String, dynamic> serializeAsMap() {
-    final Map<String, dynamic> pendingRequest = <String, dynamic>{};
-    if (this.clientMetadata != null) {
-      pendingRequest["clientMetadata"] = clientMetadata;
-    }
-    if (this.userAttributes != null) {
-      pendingRequest["userAttributes"] = userAttributes == null
-          ? null
-          : (Map.of(userAttributes!)..removeWhere((key, value) => key.readOnly))
-              .map((attribute, value) => MapEntry(attribute.key, value));
-    }
-    return pendingRequest;
-  }
+  /// A map of custom key-value pairs that you can provide as input for certain
+  /// custom workflows that this action triggers.
+  final Map<String, String>? clientMetadata;
+
+  // TODO(dnys1): Can this be removed?
```

Comment:
We will need this. Users can update attributes when confirming sign in with a new password. One use case is for users that were created in the admin console and were not given a required attribute (for example, a phone number). They can enter that attribute during confirm sign in with new password.

We support this is the authenticator, although customers have to specifiy which fields show using `ConfirmSignInNewPasswordForm.custom(fields: [ ...fields ])`

Reply:
Okay, I'll check how iOS/Android accomplishes this. It's not clear since there is no way to include user attributes in the "confirm sign in" call to Cognito. So maybe they do it separately afterwards or maybe beforehand.

Reply:
Good catch. That actually exposed a shortcoming in our library - we had previously only known which required attributes were missing by catching `InvalidParameterException`, but this information was always being returned from Cognito. Our library (or maybe the native libraries) just didn't expose it... anyway, that's a thing of the past!

https://github.com/aws-amplify/amplify-flutter-staging/pull/11

Reply:
Oh that is nice. So we can update the authenticator confirm sign in screens to display those fields by default, correct? Forcing customers to hard code a list of attributes in `ConfirmSignInNewPasswordForm.custom()` for users that were created without required attributes is definitely a poor dx.

I'm going to create a task for this.